public/app/plugins/datasource/loki/querybuilder/operationUtils.ts

```diff
@@ -187,9 +187,15 @@ export function pipelineRenderer(model: QueryBuilderOperation, def: QueryBuilder
   switch (model.id) {
     case LokiOperationId.Logfmt:
       const [strict = false, keepEmpty = false, ...labels] = model.params;
-      return `${innerExpr} | logfmt${strict ? ' --strict' : ''}${keepEmpty ? ' --keep-empty' : ''} ${labels.join(
-        ', '
-      )}`.trim();
+      return `${innerExpr} | logfmt${strict ? ' --strict' : ''}${keepEmpty ? ' --keep-empty' : ''} ${labels
+        .filter((label) => label)
+        .join(', ')}`.trim();
+    case LokiOperationId.Json:
+      return `${innerExpr} | json ${model.params.filter((param) => param).join(', ')}`.trim();
+    case LokiOperationId.Drop:
+      return `${innerExpr} | drop ${model.params.filter((param) => param).join(',')}`;
+    case LokiOperationId.Keep:
+      return `${innerExpr} | keep ${model.params.filter((param) => param).join(',')}`;
```

Comment:
Is there a reason we don't add spaces for these two operations?

Reply:
oh! nice catch 👀 . I don't think there is any reason, so I'll update it to make it more consistent.

Reply:
Fixed in https://github.com/grafana/grafana/pull/78594/commits/ead54a55c6172388f1b3db344d02f2328026a75d